updated drop_dims()

Ozan147 opened the pull request:
Fixes #135.

- With default dims = NULL, `drop_dims()` drops all single-level dimensions as expected
- `drop_dims()`'s output excludes the `draws_array` class and therefore doesn't use the `print.draws_array` method

paul-buerkner commented:
Thanks! I think we need to ensure all posterior classes are dropped if a dimension is dropped. That does mean we should handle the `draws_matrix` class as well, as well as remove the general `draws` class from the object.

Another questions is, if we still want to drop classes even if no dimension was dropped? I can think of arguments for both ways.

Ozan147 commented:
Modified the drop function to remove dimensions selectively **and/or** get rid of posterior classes. After some exploration, using an optional boolean parameter to set classes to NULL and let R decide appropriate native classes worked best for me. This also allows users to reset classes even if no dimension was dropped.

Since the function is now doing more than just dropping dimensions, I renamed it from `drop_dims()` to `drop2()` following your conventions with other functions (`stop2()`, `deparse2()`, etc.). It mirrors `base::drop()` by default but provides more flexibility with additional parameters. 

There was a separate `drop_()` helper within `rvar-slice` tools. I first tried to merge both functions but it wasn't clean due to `rvar` objects not being of `array` class and having some edge cases. Then I realized `drop_()` was a small function that was getting called only once, so I got rid of it to prevent confusion (assuming it is not used in any downstream packages as `rvar` already has a slicer method with drop option).

Let me know what you think about these changes. I can keep iterating until we are happy with it.

mjskay commented on R/rvar-slice.R:
```diff
@@ -125,7 +125,20 @@
+      new_dimnames <- .dimnames[keep_dim]
+      if (all(names(new_dimnames) == "")) names(new_dimnames) <- NULL
+      dimnames(x) <- new_dimnames
+    }
```
> There was a separate drop_() helper within rvar-slice tools. I first tried to merge both functions but it wasn't clean due to rvar objects not being of array class and having some edge cases. Then I realized drop_() was a small function that was getting called only once, so I got rid of it to prevent confusion (assuming it is not used in any downstream packages as rvar already has a slicer method with drop option).

That's fine with me. I had thought I would need that function in more than one spot in rvar internals and then I didn't. Stylistically I would probably change this:

```r
    x
  } else {
    x
  }
```

to this:

```r
  }

  x
```

Ozan147 replied:
Updated it, thanks! Let me know if there is any other change you'd like.

mjskay replied:
Thanks!

mjskay commented:
Changes on the rvar end of things look good to me, thanks!

paul-buerkner commented:
Thanks! Looks good to me too. Merging now.
